packages/components/src/autocomplete/autocompleter-ui.tsx

```diff
@@ -51,7 +58,7 @@ export function getAutoCompleterUI( autocompleter ) {
 			// eslint-disable-next-line react-hooks/exhaustive-deps
 		}, [ items ] );
 
-		if ( ! items.length > 0 ) {
+		if ( ! ( items.length > 0 ) ) {
```

Comment:
Small runtime change to allow TypeScript to parse the expression properly. Previously it was `! items.length` first, and then trying to apply the `>` operator to a boolean. This change ensures the proper order of operations.

Reply:
Filing this under "Thinks I wish I'd thought of myself"

✅ a3f56ce01f